tests/table/test_init.py

```diff
@@ -652,6 +652,58 @@ def test_update_metadata_add_snapshot(table_v2: Table) -> None:
     assert new_metadata.last_updated_ms == new_snapshot.timestamp_ms
 
 
+def test_update_metadata_create_tag(table_v2: Table) -> None:
+    update = table_v2.manage_snapshots().create_tag(snapshot_id=3051729675574597004, tag_name="test123")
+
+    new_metadata = update.table_metadata
```

Comment:
I tried committing the transaction, but that raises this error (below). I believe the api is expecting a live Noop catalog to perform the commit. 

Instead, I'm now using `update_table_metadata(table_v2.metadata, ms._updates)` that's also used in similar tests. Since the api is only changing the metadata, I think updating the table with the new metadata should be enough.
Is this ok?

error:
```
>           ms.commit()

test_init.py:660: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../pyiceberg/table/__init__.py:1877: in commit
    self._transaction._apply(*self._commit())
../../pyiceberg/table/__init__.py:302: in _apply
    self.commit_transaction()
../../pyiceberg/table/__init__.py:577: in commit_transaction
    self._table._do_commit(  # pylint: disable=W0212
../../pyiceberg/table/__init__.py:1444: in _do_commit
    response = self.catalog._commit_table(  # pylint: disable=W0212
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = NoopCatalog (<class 'pyiceberg.catalog.noop.NoopCatalog'>)
table_request = CommitTableRequest(identifier=TableIdentifier(namespace=Namespace(root=['database']), name='table'), requirements=(Ass...e='tag', snapshot_id=3051729675574597004, max_ref_age_ms=None, max_snapshot_age_ms=None, min_snapshots_to_keep=None),))

    def _commit_table(self, table_request: CommitTableRequest) -> CommitTableResponse:
>       raise NotImplementedError
E       NotImplementedError
```

Reply:
> I tried committing the transaction, but that raises this error (below)

Thanks for trying this! I overlooked that this is in unit test and we do not have a production catalog that supports `_commit_table`. How about we adding some integration tests? Those tests are under `tests/integration` folder and can be triggered via
```
make test-integration
```
In integration test environment, we have configured a RestCatalog and a HiveCatalog that can commit to a real iceberg table. The following will be a good example:
https://github.com/apache/iceberg-python/blob/cf3bf8a977f80f986237bc62293666de327871b3/tests/integration/test_partition_evolution.py#L199-L207

WDYT?

Reply:
Thank you for linking the example! I'll update my test to an integration test.